docs/advanced/shadow_prices.md

```diff
@@ -3,7 +3,26 @@
 In a linear problem, you can obtain the [shadow prices](https://en.wikipedia.org/wiki/Shadow_price) (dual variables) for each constraint from the Pyomo backend.
 This can prove particularly useful if you are linking Calliope with other models or want to dig deeper into the economic impacts of your designed energy system.
 
-You can access shadow prices for any constraint by running once you have an optimal solution by running `model.backend.shadow_prices.get("constraint_name")`, which returns an [xarray.DataArray][].
+You can access shadow prices by specifying a list of constraints for which shadow prices should be returned in the results, in your `solve` configuration:
+
+```yaml
+config:
+    solve:
+        shadow_prices: ["system_balance", ...]
+```
+
+!!! note
+
+    * Not all solvers provide access to shadow prices.
+    For instance, we know that it is possible with Gurobi and GLPK, but not with CBC.
+    Since we cannot test all Pyomo-compatible solvers, you may run into issues depending on the solver you use.
+    * You cannot access shadow prices if you have integer/binary variables in your model.
+    If you try to do so, you will receive a warning, and shadow price tracking will remain disabled.
+    * You can check the status of shadow price tracking with `model.backend.shadow_prices.is_active`.
+
+## Shadow prices when running in Python
+
+When running in Python, you can turn on shadow price tracking by running `model.backend.shadow_prices.activate()` after `model.build()`. Then, you can access shadow prices for any constraint once you have an optimal solution by running `model.backend.shadow_prices.get("constraint_name")`, which returns an [xarray.DataArray][].
```

Comment:
This makes it seem like it's the _only_ way to turn on shadow price tracking. It should be activated by default if `config.solve.shadow_prices` is set. Then it is also possible to get shadow prices in the results if calling `model.solve(shadow_prices=["system_balance", ...]`